flowpipe/utilities.py

```diff
@@ -52,3 +54,21 @@ def deserialize_graph(data):
                     upstream.outputs[plug].connect(
                         this.inputs[name][sub_plug_name])
     return graph
+
+
+class NodeEncoder(JSONEncoder):
+    """Custom JSONEncoder to handle non-json serializable node values.
+
+    If the value is not json serializable, a sha256 hash of its bytes is
+    encoded instead.
+    """
+
+    def default(self, o):
+        """Encode the object, handling type errors by encoding into sha256."""
+        try:
+            return super(NodeEncoder, self).default(o)
+        except TypeError:
+            try:
+                return sha256(bytes(o)).hexdigest()
```

Comment:
I think this should not convert to bytes as that seems to work with every object, so just:
 `return sha256(o).hexdigest()`